projects/plugins/jetpack/modules/comments/comments.php

```diff
@@ -527,19 +528,20 @@ public function watch_comment_parent() {
 	/**
 	 * Verify the hash included in remote comments.
 	 *
-	 * If the Jetpack toekn is missing we return nothing,
+	 * If the Jetpack token is missing we return nothing,
 	 * and if the token is unknown or invalid, or comments not allowed, an error is returned.
 	 *
 	 * @since JetpackComments (1.4)
-	 *
-	 * @todo We do need to add a nonce check here - internal ref for details: p1645643468937519/1645189749.180299-slack-C02HQGKMFJ8
 	 */
 	public function pre_comment_on_post() {
-		$post_array = stripslashes_deep( $_POST ); // phpcs:ignore WordPress.Security.NonceVerification.Missing
+		$post_array = stripslashes_deep( $_POST );
 
+		if ( empty( $post_array['jetpack_comments_nonce'] ) || ! wp_verify_nonce( $post_array['jetpack_comments_nonce'], "jetpack_comments_nonce-{$post_array['comment_post_ID']}" ) ) {
+			die( wp_json_encode( array( 'error' => __( 'Nonce verification failed.', 'jetpack' ) ) ) );
```

Comment:
The error here is displayed to the user, rather than used by an API client. So this should probably be more like
```suggestion
			wp_die( esc_html__( 'Nonce verification failed.', 'jetpack' ), 400 );
```
like the errors on lines 555 and 564 below.

Reply:
Good idea, thank you. Tested it out locally but I'll commit your suggestion directly.